semantic_model_generator/snowflake_utils/snowflake_connector.py

```diff
@@ -417,9 +417,18 @@ def _get_user(self) -> str:
             )
         return user
 
+    def _is_authentication_setup(self) -> bool:
+        password = env_vars.SNOWFLAKE_PASSWORD
+        private_key_file = env_vars.SNOWFLAKE_PRIVATE_KEY_FILE
```

Comment:
Do we need to be also verifying the presence of `SNOWFLAKE_PRIVATE_KEY_FILE_PWD`?